Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedListItemView.swift

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+public struct PreplannedListItemView: View {
+    /// The view model for the preplanned map.
+    @ObservedObject var preplannedMapModel: PreplannedMapModel
+    
+    /// A Boolean value indicating whether the preplanned map area can be downloaded. This is `true`
+    /// when the preplanned map area is loaded and the packaging status is `.complete`, and `false`
+    /// when the packaging status is `.processing`. If the preplanned map area has not yet loaded then
+    /// this is `nil`,
+    @State private var canDownload: Bool?
```

Comment:
There are often better types to use than `Optional<Bool>`. An enum for example or a Result (which is actually an enum).